spring-kafka/src/test/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryServiceTests.java

```diff
@@ -0,0 +1,246 @@
+import org.springframework.kafka.test.EmbeddedKafkaKraftBroker;
+import org.springframework.kafka.test.context.EmbeddedKafka;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+/**
+ * @author Soby Chacko
+ * @since 3.2.0
+ */
+@SpringJUnitConfig
+@DirtiesContext
+@EmbeddedKafka(partitions = 1,
+		topics = { "iqs-test-in", "iqs-test-out" })
+class KafkaStreamsInteractiveQueryServiceTests {
+
+	public static final String IQS_TEST_IN = "iqs-test-in";
+	public static final String IQS_TEST_OUT = "iqs-test-out";
+	public static final String STATE_STORE = "my-state-store";
+	public static final String NON_EXISTENT_STORE = "my-non-existent-store";
+
+	@Autowired
+	private EmbeddedKafkaKraftBroker embeddedKafka;
+
+	@Autowired
+	private StreamsBuilderFactoryBean streamsBuilderFactoryBean;
+
+	@Autowired
+	private KafkaTemplate<Integer, String> kafkaTemplate;
+
+	@Autowired
+	private KafkaStreamsInteractiveQueryService interactiveQueryService;
+
+	@Autowired
+	private CompletableFuture<ConsumerRecord<?, String>> resultFuture;
+
+	@Test
+	void retrieveQueryableStore() throws Exception {
+		this.kafkaTemplate.sendDefault(123, "123");
+		this.kafkaTemplate.flush();
+
+		ConsumerRecord<?, String> result = resultFuture.get(600, TimeUnit.SECONDS);
+		assertThat(result).isNotNull();
+
+		final ReadOnlyKeyValueStore<Object, Object> objectObjectReadOnlyKeyValueStore = this.interactiveQueryService
+				.retrieveQueryableStore(STATE_STORE,
+						QueryableStoreTypes.keyValueStore());
+
+		assertThat(objectObjectReadOnlyKeyValueStore.get(123)).isNotNull();
+		assertThat((Long) objectObjectReadOnlyKeyValueStore.get(123) >= 1L).isTrue();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	void retrieveNonExistentStateStoreAndVerifyRetries() throws Exception {
+		this.kafkaTemplate.sendDefault(123, "123");
+		this.kafkaTemplate.flush();
+
+		ConsumerRecord<?, String> result = resultFuture.get(600, TimeUnit.SECONDS);
+		assertThat(result).isNotNull();
+
+		assertThat(this.streamsBuilderFactoryBean.getKafkaStreams()).isNotNull();
+		final KafkaStreams kafkaStreams = spy(this.streamsBuilderFactoryBean.getKafkaStreams());
+		assertThat(kafkaStreams).isNotNull();
+
+		Field kafkaStreamsField = KafkaStreamsInteractiveQueryService.class.getDeclaredField("kafkaStreams");
+		kafkaStreamsField.setAccessible(true);
+		kafkaStreamsField.set(interactiveQueryService, kafkaStreams);
+
+		Throwable exc = null;
+		try {
+			this.interactiveQueryService
+					.retrieveQueryableStore(NON_EXISTENT_STORE, QueryableStoreTypes.keyValueStore());
+		}
+		catch (Exception e) {
+			exc = e;
```

Comment:
I prefer to use `assertThatException()` instead of `try..catch`: must better reporting.